presto-orc/src/test/java/com/facebook/presto/orc/TestDwrfStripeCaching.java

```diff
@@ -331,4 +352,66 @@ public StripeMetadataSource create(Optional<DwrfStripeCache> dwrfStripeCache)
             return dwrfStripeCache;
         }
     }
+
+    private void assertFileContent(File orcFile)
+            throws IOException
+    {
+        // all files have 400 rows (4 stripes with 100 rows each) with the following values:
+        // column 0 (int) = i
+        // column 1 (int) = Integer.MAX_VALUE
+        // column 2 (int) = i * 10
+        try (OrcDataSource orcDataSource = new FileOrcDataSource(
```

Comment:
Can we wrap the FileOrcDataSource and verify that, when using StripeMetadatasource with stripe cache the reads are happening accordingly. When stripe cache is disabled the positions are read correctly.

Reply:
Added test cases verifying that we don't read disk ranges for stripe indexes and footers when the stripe cache is enabled. I'm not sure that doing the same with disabled stripe cache will make sense, because there would be no component returning the data from the cache.

What I can do is to add test cases to make sure that we read file correctly (row value wise) when caching is disabled. Does that sound good?

Reply:
Added asserts to verify file content when we pass in DwrfAwareStripeMetadataSource, but disable DWRF stripe cache feature.

Reply:
The new code change looks good to me.